config/jobs/image-pushing/k8s-staging-dns.yaml

```diff
@@ -0,0 +1,32 @@
+postsubmits:
+  # this is the github repo we'll build from; this block needs to be repeated for each repo.
+  kubernetes/dns:
+    - name: dns-push-images
+      cluster: test-infra-trusted
+      annotations:
+        # this is the name of some testgrid dashboard to report to.
+        testgrid-dashboards: sig-network-dns
+      decorate: true
+      branches:
+        - ^master$
+      spec:
+        containers:
```

Comment:
I think you can just use a `serviceAccountName` like in https://github.com/kubernetes/test-infra/blob/master/config/jobs/image-pushing/k8s-staging-artifact-promoter.yaml#L13. Not sure if the TODO comment there is applicable @fejta

Reply:
Done

Reply:
Yes, the image pusher should use the pusher, but I believe everything is erroneously granted to the deployer right now (which is supposed to be used to change k8s resources)

So keep the TODO

Reply:
Thanks Erick. I have made the change.